websub-ballerina/subscriber_service_endpoint.bal

```diff
@@ -276,11 +278,37 @@ public type ExtensionConfig record {|
 #
 # + resourceUrl - The resource URL advertising the hub and topic URLs
 # + publisherClientConfig - The configuration for the publisher client
+# + expectedMediaType - The expected media type for the subscriber client
+# + expectedLanguageType - The expected language type for the subscriber client
 # + return - A `(hub, topic)` as a `(string, string)` if successful or else an `error` if not
-function retrieveHubAndTopicUrl(string resourceUrl, http:ClientConfiguration? publisherClientConfig)
+function retrieveHubAndTopicUrl(string resourceUrl, http:ClientConfiguration? publisherClientConfig, string?|string[] expectedMediaType, string?|string[] expectedLanguageType)
         returns @tainted [string, string]|error {
     http:Client resourceEP = new http:Client(resourceUrl, publisherClientConfig);
     http:Request request = new;
+    if (expectedMediaType is string) {
+        request.addHeader(ACCEPT_HEADER, expectedMediaType);
+    }
+    
+    if (expectedMediaType is string[]) {
+        string acceptMeadiaTypesString = expectedMediaType[0];
+        foreach string mediaType in expectedMediaType {
+            acceptMeadiaTypesString = acceptMeadiaTypesString.concat(", ", mediaType);
+        }
```

Comment:
Here the first element of the array is getting appended 2 times to the `acceptMeadiaTypesString`. 
First when you initiate the first element is assigned to the `acceptMeadiaTypesString` and second within the foreach loop, the first element getting appended to `acceptMeadiaTypesString`.
Shall we fix it?